services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,19 +43,52 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  if (event == NULL) return ERR_CODE_INVALID_ARG;  
+  if (xQueueSend(thermalMgrQueueHandle, event, 0) == errQUEUE_FULL) return ERR_CODE_QUEUE_FULL;
+  
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t event;
+  event.type = THERMAL_MGR_EVENT_INTERRUPT;
+  thermalMgrSendEvent(&event);
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+  // if (pvParameters == NULL) return ERR_CODE_INVALID_ARG;
+
+  error_code_t errCode;
+  thermal_mgr_event_t event; // must be of type THERMAL_MGR_EVENT_MEASURE_TEMP_CMD
+  float tempC;
```

Comment:
in general try to limit the scope of your variables to as small is possible, so declare these within the while loop